sql/src/main/java/io/crate/planner/operators/Union.java

```diff
@@ -195,21 +191,4 @@ public long estimatedRowSize() {
     public <C, R> R accept(LogicalPlanVisitor<C, R> visitor, C context) {
         return visitor.visitUnion(this, context);
     }
-
-    /**
-     * Wraps the plan inside a Merge plan if limit or offset need to be applied.
-     */
-    private static ExecutionPlan addMergeIfNeeded(ExecutionPlan plan, PlannerContext plannerContext) {
-        ResultDescription resultDescription = plan.resultDescription();
-        if (resultDescription.hasRemainingLimitOrOffset()) {
-            // Do a merge because we have to apply a limit/offset projection
-            //
-            // Note: Currently, this is performed on the handler node. It would be possible to
-            // do this on another involved node instead but we don't do that for now because
-            // the Merge of the union itself is always performed on the handler. So the
-            // performance gain would be small.
-            return Merge.ensureOnHandler(plan, plannerContext);
```

Comment:
This removed the `resultDescription.orderBy` information, which is required so that further merges on top of the union are ordered correctly.

The `UnionExecutionPlan` carries on the `limit + offset`, so that later operators which force a merge apply it correctly.

Reply:
@mkleen had to push a fixup because if there is a limit - order by operator beneath the union   it must be merged before the union. So I had to re-add this part